### ISSUE TITLE
Make hook prefixes consistent

### ISSUE THREAD

**Issue by GaryJones** [Parsely/wp-parsely#169]
The current non-core filters (there are no action hooks) in the plugin are:

 - `wp_parsely_post_tags`
 - `wp_parsely_post_category`
 - `wp_parsely_pre_authors`
 - `wp_parsely_post_authors`
 - `wp_parsely_permalink`
 - `after_set_parsely_page`
 - `parsely_filter_insert_javascript`

For version 2.5.0, the bottom two should have a new filter hook with the consistent `wp_parsely` prefix, and change the inconsistent ones from `apply_filters()` to [`apply_filters_deprecated()`](https://developer.wordpress.org/reference/functions/apply_filters_deprecated/).

In 3.0.0, the old filters can be removed. Any customer using the old filter will need to update their code to reference the new filter name instead. Once a PR is merged for this issue, then a new issue / PR for the removal should be created.

We can protect against a re-introduction of incorrectly prefixed hook names by setting a [prefix for all globals](https://github.com/WordPress/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#naming-conventions-prefix-everything-in-the-global-namespace) in a PHPCS config file.
